extensions/wikia/Search/classes/IndexService/All.php

```diff
@@ -3,54 +3,60 @@
  * Class definition for \Wikia\Search\IndexService\All
  */
 namespace Wikia\Search\IndexService;
+
 use Wikia\Search\MediaWikiService;
+
 /**
  * Aggregates all other services into a single request -- good for populating a full index
+ *
  * @author relwell
  * @package Search
  * @subpackage IndexService
  */
-class All extends AbstractService
-{
+class All extends AbstractService {
 	/**
-	 * These are the services whose outputs will be aggregated during execute. 
+	 * These are the services whose outputs will be aggregated during execute.
+	 *
 	 * @var array
 	 */
-	protected $services = array(
-			'Wikia\Search\IndexService\DefaultContent'  => null,
-			'Wikia\Search\IndexService\BacklinkCount'   => null,
-			'Wikia\Search\IndexService\MediaData'       => null,
-			'Wikia\Search\IndexService\Metadata'        => null,
-			'Wikia\Search\IndexService\Redirects'       => null,
-			'Wikia\Search\IndexService\Wam'             => null,
-			'Wikia\Search\IndexService\WikiPromoData'   => null,
-			'Wikia\Search\IndexService\WikiStats'       => null,
-			'Wikia\Search\IndexService\WikiViews'       => null,
-			'Wikia\Search\IndexService\VideoViews'      => null, // note the order of operations -- AFTER metadata
-			'Wikia\Search\IndexService\ArticleQuality'  => null
-			);
-	
+	protected $services = [
+		'Wikia\Search\IndexService\DefaultContent' => null,
+		'Wikia\Search\IndexService\BacklinkCount' => null,
+		'Wikia\Search\IndexService\MediaData' => null,
+		'Wikia\Search\IndexService\Metadata' => null,
+		'Wikia\Search\IndexService\Redirects' => null,
+		'Wikia\Search\IndexService\Wam' => null,
+		'Wikia\Search\IndexService\WikiPromoData' => null,
+		'Wikia\Search\IndexService\WikiStats' => null,
+		'Wikia\Search\IndexService\WikiViews' => null,
+		'Wikia\Search\IndexService\VideoViews' => null, // note the order of operations -- AFTER metadata
+		'Wikia\Search\IndexService\ArticleQuality' => null
+	];
+
 	/**
 	 * Invokes a bunch of other services' execute functions
+	 *
+	 * @throws \Exception
 	 * @see \Wikia\Search\IndexService\AbstractService::execute()
 	 * @return array
 	 */
 	public function execute() {
 		if ( $this->currentPageId === null ) {
 			throw new \Exception( "This service requires a page ID to be set." );
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Define and throw a dedicated exception instead of using a generic one. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS112)